File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/02-IterativeAlgorithms/LeetCode-0771-2.py
class Solution:
    # O(n+m) time algorithm, where n & m are the lengths of strings jewels and stones
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        # A big enough table to contain both upper & lower-case letters
        LUT = [False]*(ord('z')+1)

        # Mark all Jewels in the table
        for ch in jewels: 
            LUT[ord(ch)] = True

        # Go over stones. If a letter is a jewel, add it to the count
        count = 0
        for ch in stones:
            if LUT[ord(ch)]: 
                count += 1
        return count

File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/02-IterativeAlgorithms/LeetCode-0771-2.cpp
class Solution {
public:
    ///------------------------------------------------------------------
    /// O(n+m) time algorithm, where n & m are the lengths of strings jewels and stones
    ///
    int numJewelsInStones(string jewels, string stones) {
        // A big enough table to contain both upper & lower-case letters
        #define NO_LETTERS 'z'+ 1
        bool LUT[NO_LETTERS] = {false};

        // Mark all Jewels in the table
        for (auto ch: jewels) LUT[ch] = true;

        // Go over stones. If a letter is a jewel, add it to the count
        int count = 0;
        for (auto ch: stones){
            if (LUT[ch]) count++;
        } //end-if

        return count;
    } // end-numJewelsInStones
};


